src/ServiceControl/Monitoring/HeartbeatEndpointSettingsSyncHostedService.cs

```diff
@@ -0,0 +1,126 @@
+﻿namespace ServiceControl.Monitoring;
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Persistence;
+using ServiceBus.Management.Infrastructure.Settings;
+
+public class HeartbeatEndpointSettingsSyncHostedService(
+    IMonitoringDataStore monitoringDataStore,
+    IEndpointSettingsStore endpointSettingsStore,
+    IEndpointInstanceMonitoring endpointInstanceMonitoring,
+    Settings settings,
+    TimeProvider timeProvider,
+    ILogger<HeartbeatEndpointSettingsSyncHostedService> logger)
+    : BackgroundService
+{
+    public TimeSpan DelayStart { get; set; } = TimeSpan.FromSeconds(20);
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        logger.LogInformation($"Starting {nameof(HeartbeatEndpointSettingsSyncHostedService)}");
+
+        try
+        {
+            await Task.Delay(DelayStart, timeProvider, cancellationToken);
```

Comment:
why?

Reply:
we try to stagger all the background tasks so not all start at the same time

Reply:
so each task needs to know how much every other task is staggering itself by? Shouldn't there be an overall controller for that? Or a service where it has a .GetNextAvailableStaggeredDelay function?